### PR TITLE
[IMP] payment, *: update urls to the new format

### PR DESCRIPTION
**Pull request by sami-odoo**
\* = [payment_stripe, point_of_sale, project, project_todo, purchase, purchase_product_matrix]

This commit modifies existing URLs to the new format so that the URLs become more human-readable.

Task-3820230

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase_stock/tests/test_replenish_wizard.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def test_replenish_buy_1(self):
 
         last_po_id = False
         if purchase_order_id and model_name:
-            last_po_id = self.env[model_name[0]].browse(int(purchase_order_id[0]))
+            last_po_id = self.env[model_name].browse(int(purchase_order_id))
```
Comment (jpp-odoo) [inline addons/purchase_stock/tests/test_replenish_wizard.py:57]: Not related to the PR (maybe do it in a separate PR or commit)

File: addons/stock/models/stock_orderpoint.py [hunk 2]
```diff
@@ -525,7 +525,7 @@ def _get_replenishment_order_notification(self):
                     'message': '%s',
                     'links': [{
                         'label': move.picking_id.name,
-                        'url': f'/web#action={action.id}&id={move.picking_id.id}&model=stock.picking&view_type=form'
+                        'url': f'/odoo/action-{action}/{move.picking_id.id}'
```
Comment (jpp-odoo) [inline addons/stock/models/stock_orderpoint.py:528]:
```suggestion
                        'url': f'/odoo/action-stock.stock_picking_action_picking_type/{move.picking_id.id}'
```

File: addons/stock/tests/test_picking_tours.py [hunk 3]
```diff
@@ -15,8 +15,8 @@ def setUp(self):
         return super().setUp()
 
     def _get_picking_url(self, picking_id):
-        action = self.env["ir.actions.actions"]._for_xml_id("stock.action_picking_tree_incoming")
-        return '/web#action=%s&id=%s&model=stock.picking&view_type=form' % (action['id'], picking_id)
+        action = "stock.action_picking_tree_incoming"
+        return '/odoo/action-%s/%s' % (action, picking_id)
```
Comment (jpp-odoo) [inline addons/stock/tests/test_picking_tours.py:19]:
```suggestion
        return '/odoo/action-stock.action_picking_tree_incoming/%s' % (picking_id)
```

File: addons/stock/wizard/product_replenish.py [hunk 4]
```diff
@@ -143,10 +143,10 @@ def _get_record_to_notify(self, date):
 
     def _get_replenishment_order_notification_link(self, move):
         if move.picking_id:
-            action = self.env.ref('stock.stock_picking_action_picking_type')
+            action = 'stock.stock_picking_action_picking_type'
             return [{
                 'label': move.picking_id.name,
-                'url': f'/web#action={action.id}&id={move.picking_id.id}&model=stock.picking&view_type=form'
+                'url': f'/odoo/action-{action}/{move.picking_id.id}'
```
Comment (jpp-odoo) [inline addons/stock/wizard/product_replenish.py:149]:
```suggestion
                'url': f'/odoo/action-stock.stock_picking_action_picking_type/{move.picking_id.id}'
```

**Review by jpp-odoo — CHANGES_REQUESTED** [review]

**Review by AntoineVDV — APPROVED** [review]

**Review by adgu-odoo — APPROVED** [review]
Pos part is good for me

**Comment by jpp-odoo** [comment]
@robodoo r+
